### PR TITLE
feat: [sc-15138] public API filter, list and tag dtos

### PR DESCRIPTION
**Pull request by bitpetr**
Story details: https://app.shortcut.com/systeme/story/15138

### CODE REVIEW

**Comment by bitpetr** [comment]
Public API endpoints: https://github.com/systemeio/public-api/pull/9
Inner API endpoints: https://github.com/systemeio/backend/pull/7836

File: src/TagInputDto.php [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Systeme\Component\Dto;
+
+class TagInputDto
+{
+    private string $name;
```
Comment (osipal) [inline src/TagInputDto.php:9]: Do you think that the property should be declared as nullable and initialized?
```php
private ?string $name = null;
```

Same question for classes `ListFilterDto`, `ListSearchFilterDto`

Comment (bitpetr) [reply src/TagInputDto.php:9]: I'm honestly confused, the style in the repo is all over the place.
Right now, we don't need nullable properties. Logically, `name` should not be null and normalizers work fine with setters.

What are the arguments for using null-initiliazed constructor properties? :thinking:

Comment (osipal) [reply src/TagInputDto.php:9]: What I mean is this: if you create an object and call the "getName()" getter before calling the setter, there will be an error.
But we won't explicitly create objects, so forget it, I'll close the discussion.

**Review by osipal — APPROVED** [review]

**Review by KoNekoD — APPROVED** [review]

**Pull request merged**